File: Android/app/src/main/java/com/felipearpa/tyche/network/LocalUrlBasePathProvider.kt
package com.felipearpa.tyche.network

import com.felipearpa.tyche.BuildConfig
import com.felipearpa.tyche.core.network.UrlBasePathProvider

class LocalUrlBasePathProvider : UrlBasePathProvider {
    override val basePath: String = BuildConfig.URL_BASE_BATH
}

File: iOS/Tyche/Tyche/Config/LocalURLBasePathProvider.swift
import Foundation
import Core

public class LocalURLBasePathProvider : URLBasePathProvider {
    private let basePath: String

    public convenience init() {
        let basePath = (Bundle.main.object(forInfoDictionaryKey: "URL_BASE_PATH") as? String) ?? ""
        self.init(basePath: basePath)
    }

    init(basePath: String) {
        if basePath.hasSuffix("/") {
            self.basePath = basePath
        } else {
            self.basePath = basePath + "/"
        }
    }

    public func prependBasePath(_ string: String) -> URL? {
        return URL(string: basePath + string)
    }
}
